library/system/src/lib/yast2/execute.rb

```diff
@@ -87,11 +104,119 @@ def self.locally(*args)
     # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
     # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
     # @raise Cheetah::ExecutionFailed if the command fails
-    def self.locally!(*args)
-      Cheetah.run(*args)
+    def locally!(*args)
+      run_or_chain(args)
+    end
+
+    # Runs a command described by *args* and returns its output
+    #
+    # It also globally switches the default Cheetah logger to
+    # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
+    #
+    # @param args [Array<Object>] see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+    # @return [String] command output or an empty string if the command fails.
+    def stdout(*args)
+      chaining_object(yast_stdout: true, stdout: :capture).run_or_chain(args)
+    end
+
+  protected
+
+    # Decides either to run the command or to chain the call in case that no argmuments
+    # are given.
+    #
+    # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+    # @return [Object, ExecuteClass] result of running the command or a chaining object.
+    def run_or_chain(args)
+      args.none? ? self : run(*args)
+    end
+
+  private
+
+    # Options to add when running a command
+    #
+    # Some options are intended to control the behavior and they are not passed to
+    # Cheetah.run. For example:
+    #
+    # * `yast_popup`: to indicate whether a popup should be shown when the command fails.
+    # * `yast_stdout`: to indicate whether the command always should return an output,
+    #     even when it fails.
+    #
+    # @return [Hash<Symbol, Object>]
+    attr_reader :options
+
+    # New object to chain method calls
+    #
+    # The new object contains current object options plus given new options.
+    #
+    # @param new_options [Hash<Symbol, Object>]
+    # @return [ExecuteClass]
+    def chaining_object(new_options)
+      self.class.new(options.merge(new_options))
+    end
+
+    # Runs the given command
+    #
+    # It takes into account the object options when running the command.
+    # Note that `yast_popup` takes precedence over `yast_stdout`. So, when both options
```

Comment:
please mention also that if popup or stdout is used, that `!` variant does not raise. so `locally!.stdout` does not raise exception.